ray-operator/apis/ray/v1/rayjob_types.go

```diff
@@ -119,6 +119,15 @@ type RayJobSpec struct {
 	// If the RayCluster is already created, it will be deleted.
 	// In case of transition to false a new RayCluster will be created.
 	Suspend bool `json:"suspend,omitempty"`
+	// ManagedBy is used to indicate the controller or entity that manages a RayJob.
+	// The value must be either empty, 'ray.io/kuberay-operator' or 'kueue.x-k8s.io/multikueue'.
+	// The kuberay-operator reconciles a RayJob which doesn't have this field at all or
+	// the field value is the reserved string 'ray.io/kuberay-operator',
+	// but delegates reconciling the RayJob with 'kueue.x-k8s.io/multikueue' to the Kueue.
+	// The field is immutable.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="the managedBy field is immutable"
+	// +kubebuilder:validation:XValidation:rule="self in ['', 'ray.io/kuberay-operator', 'kueue.x-k8s.io/multikueue']",message="the managedBy field value must be either empty, 'ray.io/kuberay-operator' or 'kueue.x-k8s.io/multikueue'"
```

Comment:
managedBy seems like it may have use-cases beyond Kueue, is it necessary to restrict valid values to only ray.io/kuberay-operator and kueue.x-k8s.io/multikueue?

Reply:
Yes, it is a more generic concept, we could relax the validation - this is what for example batch/Job and JobSet do. 

OTOH Kubeflow decided to be more restrictive and only allow for MultiKueue (as in the proposal here).

The reasoning for strict validation is:
1. MultiKueue is the only use case for now
2. it is easier to relax validation than strenghten in the future releases
3. external controllers might have bugs which could be problematic for OnCall

The strategy which batch/Job adopts to counter (3.) is very strict validation of status updates (see [KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/4368-support-managed-by-for-batch-jobs#job-status-validation)), but this is an extra effort (see [validation of Job status in k8s](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/batch/validation/validation.go#L459-L606)). 

With the validation for .status we can give early feedback to the authors of the external controllers, so that most bugs can be caught early and eliminated. TBH fitting an external controller within the same API is unlikely to happen, so I'm not sure there will be other external controllers, but you never know.

Reply:
okay that makes sense to me, we can always remove it if people ask for it